applications/FemToDemApplication/custom_processes/update_dem_kinematics_process.cpp

```diff
@@ -0,0 +1,89 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics FemDem Application
+//
+//  License:		 BSD License
+//					 Kratos default license:
+//kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//
+
+#include "custom_processes/update_dem_kinematics_process.h"
+#include "includes/define.h"
+#include "includes/kratos_flags.h"
+
+namespace Kratos {
+
+UpdateDemKinematicsProcess::UpdateDemKinematicsProcess(
+    ModelPart& rModelPart,
+    ModelPart& rDemModelPart)
+    : mrModelPart(rModelPart),
+      mrDEMModelPart(rDemModelPart)
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void UpdateDemKinematicsProcess::Execute() 
+{
+    const auto it_node_begin = mrModelPart.NodesBegin();
+    //#pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(mrModelPart.Nodes().size()); i++) {
+        auto it_node = it_node_begin + i;
+        if (it_node->GetValue(IS_DEM)) {
+            const int node_id = it_node->Id();
+            auto& associated_dem = mrDEMModelPart.GetNode(node_id);
```

Comment:
This is slow...

Reply:
Is the node shared?

Reply:
It is not shared... There are some cases that could exist a DEM node but the original fem node is erased. Maybe I can rethink this issue in the future :S